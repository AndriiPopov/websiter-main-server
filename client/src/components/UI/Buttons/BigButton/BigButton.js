import React from 'react';
import PropTypes from 'prop-types';

import classes from './BigButton.module.css';
import Svg from '../../../Svg/Svg';

export const BigButton = props => {
    return (
        <button 
            className={classes.Button}
            onClick={props.buttonClicked}>
            <table className={classes.Table}>
                <tbody>
                    <tr>
                        <td><Svg icon={props.icon}/></td>
                    </tr>
                    <tr>
                        <td>{props.title}</td>
                    </tr>
                </tbody>
            </table>
        </button>
    );
};

BigButton.propTypes = {
    title: PropTypes.string,
    buttonClicked: PropTypes.func,
    icon: PropTypes.string
};

export default BigButton;