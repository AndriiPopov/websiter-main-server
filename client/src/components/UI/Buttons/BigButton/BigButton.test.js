import React from 'react';
import {mount, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { BigButton } from './BigButton';

configure({adapter: new Adapter()});

describe('BigButton', () => {
    it('should be defined', () => {
        expect(BigButton).toBeDefined();
    });

    it('it should render button with onclick null when disabled', () => {
        const tree = mount(
            <BigButton 
                icon='<svg height="16" viewBox="0 0 24 24" width="16"><path d="M0 0h24v24H0z" fill="none"></path><path d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"></path></svg>' 
                title="Button"
                buttonClicked="clickOnButton" />
        );
        expect(toJson(tree)).toMatchSnapshot();
    });
});