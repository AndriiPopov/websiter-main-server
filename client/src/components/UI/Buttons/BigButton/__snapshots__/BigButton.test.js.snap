// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigButton it should render button with onclick null when disabled 1`] = `
<BigButton
  buttonClicked="clickOnButton"
  icon="<svg height=\\"16\\" viewBox=\\"0 0 24 24\\" width=\\"16\\"><path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path><path d=\\"M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z\\"></path></svg>"
  title="Button"
>
  <button
    className="Button"
    onClick={[Function]}
  >
    <table
      className="Table"
    >
      <tbody>
        <tr>
          <td>
            <Svg
              icon="<svg height=\\"16\\" viewBox=\\"0 0 24 24\\" width=\\"16\\"><path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path><path d=\\"M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z\\"></path></svg>"
            >
              <svg
                height="16"
                viewBox="0 0 24 24"
                width="16"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  key="0"
                />
                <path
                  d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                  key="1"
                />
              </svg>
            </Svg>
          </td>
        </tr>
        <tr>
          <td>
            Button
          </td>
        </tr>
      </tbody>
    </table>
  </button>
</BigButton>
`;
